assets/js/admin/orders.js

```diff
@@ -11,4 +11,91 @@ jQuery( document ).ready( ( $ ) => {
 
 	'use strict';
 
+	const isCommerceOrder = Boolean( wc_facebook_commerce_orders.is_commerce_order );
+
+	window.WCFacebookCommerceOrderOperations = {
+		/**
+		 * Restrict order status options to only allowed options.
+		 *
+		 * @param {Object} $orderStatus Order select jQuery DOM object
+		 */
+		restrict_order_statuses             : $orderStatus => {
+
+			$orderStatus.find( 'option' ).each( function ( index, option ) {
+
+				// check if option value in the allowed list or not
+				if ( wc_facebook_commerce_orders.allowed_commerce_statuses.indexOf( option.value ) === -1 ) {
+					// delete/remove option if not allowed
+					option.remove();
+				}
+			} );
+		},
+		/**
+		 * Enable or Disable order created fields.
+		 *
+		 * @param {Boolean} enable wither to enable date fields (true) or not (false)
+		 */
+		toggle_created_date_fields_status   : enable => {
+			$( '#order_data' ).find( 'input[name*=order_date]' ).prop( 'disabled', !enable ).toggleClass( 'disabled', !enable );
+		},
+		/**
+		 * Disable order status field
+		 *
+		 * @param {Object} $orderStatus Order select jQuery DOM object
+		 */
+		disable_order_status_field          : ( $orderStatus ) => {
+			$orderStatus.prop( 'disabled', true ).addClass( 'disabled' );
+		},
+		/**
+		 * Toggle customer field
+		 *
+		 * @param {Boolean} hide
+		 */
+		toggle_order_customer_field         : ( hide ) => {
+			$( '#order_data' ).find( '.form-field.wc-customer-user' ).addClass( 'hidden', hide );
+		},
+		/**
+		 * Toggle customer field
+		 *
+		 * @param {Boolean} hide
+		 */
+		toggle_billing_and_shipping_fields  : ( hide ) => {
+			$( '#order_data' ).find( 'a.edit_address' ).addClass( 'hidden', hide );
+		},
+		/**
+		 * Disable order status field
+		 *
+		 * @param {Object} $orderStatus Order select jQuery DOM object
+		 */
+		disable_pending_order_related_fields: ( $orderStatus ) => {
+
+			WCFacebookCommerceOrderOperations.toggle_created_date_fields_status( false );
+			WCFacebookCommerceOrderOperations.disable_order_status_field( $orderStatus );
+			WCFacebookCommerceOrderOperations.toggle_order_customer_field( true );
+			WCFacebookCommerceOrderOperations.toggle_billing_and_shipping_fields( true );
+		}
+	};
+
+	if ( isCommerceOrder ) {
+
+		let $orderStatus = $( '#order_status' );
+
+		WCFacebookCommerceOrderOperations.restrict_order_statuses( $orderStatus );
+
+		if ( 'pending' === wc_facebook_commerce_orders.order_status ) {
+			WCFacebookCommerceOrderOperations.disable_pending_order_related_fields( $orderStatus );
+		}
+
+		if ( 'cancelled' === wc_facebook_commerce_orders.order_status ) {
+			WCFacebookCommerceOrderOperations.disable_order_status_field( $orderStatus );
+		}
+
+		$orderStatus.on( 'change', () => {
+
+			if ( 'wc-pending' === $orderStatus.val() ) {
```

Comment:
If we are evaluating the change to `wc-pending` in real time (on the change event), should we not do the same for the `wc-cancelled`?

Reply:
I guess you are right, we don't need the on-change event handler as that would prevent the data saving upon order update ... I will remove it

Reply:
I believe it would be good to double check (with Chase or perhaps Lindsey) if we expect the fields to be blocked on change as well or not.

Reply:
Based on another story which hides the update button meta box if order status is pending, I believe that is the case